op-service/txmgr/txmgr.go

```diff
@@ -320,6 +331,46 @@ func (m *SimpleTxManager) craftTx(ctx context.Context, candidate TxCandidate) (*
 	return m.signWithNextNonce(ctx, txMessage) // signer sets the nonce field of the tx
 }
 
+func (m *SimpleTxManager) GetMinBaseFee() *big.Int {
+	return m.cfg.MinBaseFee
+}
+
+func (m *SimpleTxManager) SetMinBaseFee(val *big.Int) {
+	m.cfg.MinBaseFee = val
+}
```

Comment:
I'm wondering if these functions are thread safe -- I am not sure if the txManager has other threads that might be running and using these values. If so, there's a potential for thread-unsafe updates to these values.

Reply:
These functions are not currently thread safe. `txmgr.Send()` is called in a go routine [here](https://github.com/ethereum-optimism/optimism/blob/develop/op-service/txmgr/queue.go#L60-L65). Would we want to preserve all cfg values (copy a snapshot of `cfg` within `txmgr.Send()` and pass those values to underlying functions) such that they can't be altered for a given go routine after its started (values would only be changed for future `txmgr.Send() invocations), or would a simple mutex suffice? 

Simple mutex added here for reference: https://github.com/ethereum-optimism/optimism/pull/10897/commits/988e78cbebd4d6a8248d82a1d288a441a3d1298e

Reply:
Related: would we want to create a `txmgr_setGasParams` rpc method to grab the mutex and set all gas params in a single request to reduce potential of unintended configurations?

Reply:
Will not implement the rpc method to set multiple params at once